src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/EditorConfigFileGenerator_NamingStyles.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles;
+
+namespace Microsoft.CodeAnalysis.Options
+{
+    internal static partial class EditorConfigFileGenerator
```

Comment:
All code moved as-is from Workspaces down to shared layer from a different file.